cvat-canvas/src/typescript/groupHandler.ts

```diff
@@ -1,273 +1,75 @@
 // Copyright (C) 2019-2022 Intel Corporation
-// Copyright (C) 2022 CVAT.ai Corporation
+// Copyright (C) 2022-2023 CVAT.ai Corporation
 //
 // SPDX-License-Identifier: MIT
 
-import * as SVG from 'svg.js';
 import { GroupData } from './canvasModel';
-import { expandChannels, imageDataToDataURL, translateToSVG } from './shared';
+import { ObjectSelector, SelectionFilter } from './objectSelector';
 
 export interface GroupHandler {
-    group(groupData: GroupData): void;
+    group(groupData: GroupData, selectionFilter: SelectionFilter): void;
     select(state: any): void;
     cancel(): void;
-    resetSelectedObjects(): void;
 }
 
 export class GroupHandlerImpl implements GroupHandler {
-    // callback is used to notify about grouping end
-    private onGroupDone: (objects?: any[]) => void;
-    private getStates: () => any[];
-    private onFindObject: (event: MouseEvent) => void;
-    private bindedOnSelectStart: (event: MouseEvent) => void;
-    private bindedOnSelectUpdate: (event: MouseEvent) => void;
-    private bindedOnSelectStop: (event: MouseEvent) => void;
-    private selectionRect: SVG.Rect | null;
-    private startSelectionPoint: {
-        x: number;
-        y: number;
-    };
-    private canvas: SVG.Container;
+    private onSelectDone: (objects?: any[], duration?: number) => void;
```

Comment:
Are `objects` here a ObjectState[]?
The same for cvanvasView 396, 421

Reply:
yes, but object states are not typed in this module